src/test/ismine_tests.cpp

```diff
@@ -216,4 +216,46 @@ BOOST_AUTO_TEST_CASE(is_stakeable_by_me_remote_staking_watchonly) {
   BOOST_CHECK(!IsStakeableByMe(keystore, rsp2wsh));
 }
 
+BOOST_AUTO_TEST_CASE(is_stakeable_by_me_destinations) {
+
+  // Test that the P2SH output type is NOT stakeable
+  {
+    CKey key;
+    key.MakeNewKey(true);
+    CBasicKeyStore keystore;
+    keystore.AddKey(key);
+
+    CTxDestination dest = GetDestinationForKey(key.GetPubKey(), OutputType::P2SH_SEGWIT);
+    CScript script = GetScriptForDestination(dest);
+
+    BOOST_CHECK(!IsStakeableByMe(keystore, script));
+  }
+
+  // Test that the legacy output type is stakeable
+  {
+    CKey key;
+    key.MakeNewKey(true);
+    CBasicKeyStore keystore;
+    keystore.AddKey(key);
+
+    CTxDestination dest = GetDestinationForKey(key.GetPubKey(), OutputType::LEGACY);
```

Comment:
Do we actually want these to be stakeable?

Reply:
At the moment yes, I will come with a follow up PR that is removing suppor to stake with P2PKH, then I'll also change this test.